pkg/sriov/sriov.go

```diff
@@ -0,0 +1,113 @@
+//
+// Use and distribution licensed under the Apache license version 2.
+//
+// See the COPYING file in the root project directory for full text.
+//
+
+package sriov
+
+import (
+	"fmt"
+
+	"github.com/jaypipes/ghw/pkg/context"
+	"github.com/jaypipes/ghw/pkg/marshal"
+	"github.com/jaypipes/ghw/pkg/option"
+	"github.com/jaypipes/ghw/pkg/pci"
+	pciaddr "github.com/jaypipes/ghw/pkg/pci/address"
+)
+
+type Device struct {
+	Interfaces []string `json:"interfaces"`
+	// the PCI address where the SRIOV instance can be found
+	Address *pciaddr.Address `json:"address"`
+	PCI     *pci.Device      `json:"pci"`
+}
```

Comment:
Instead of creating a new `pkg/sriov` package and `pkg/sriov.Device` struct` I think we should just modify the `pkg/pci.Device` struct to read SR-IOV (and network interface) information when the PCI device is a PF or VF.

We should add a new `Interface` struct to `pkg/net`:

```go
// Interface represents a physical or virtual network interface on the host
type Interface struct {
    // Name is the name of the interface on the host
    Name string `json:"name"`
}
```

We can then modify the `pkg/pci.Device` struct to add the following:

```go
...
import (
...
    "github.com/jaypipes/ghw/pkg/net"
...
)
...
type Device struct {
...
    // NetworkInterfaces is the collection of network interfaces housed on this
    // PCI device.
    NetworkInterfaces []net.Interface `json:"network_interfaces,omitempty"`
}
```

Reply:
This is an interesting idea. I'd like to process the concept a bit more to fully grok it but doesn't look bad at all. However this also mean that we aim to absorb all the PCI device type (= gpu) into the PCI package in the long run?
Or it's just SRIOV which is not special enough? (again no strong feeling, but wondering about the general direction)